### ISSUE TITLE
feature request the ability to update epubs allready downloaded?

### ISSUE THREAD

**Issue by mhussaincov93** [dteviot/WebToEpub#812]
would it be possible to have the ability to update epubs that have allready been downloaded?
I believe this would be very useful
Majid

**Comment by dteviot** [comment]
@mhussaincov93 
More details please.  Specifically, how does this differ from "Include in reading list" in the Advanced Options?

**Comment by budikesuma** [comment]
It should be closed immediately since the request is available in W2E's advance settings in the `reading list` feature.

**Comment by vanessabluellie** [comment]
@dteviot 

Hello! I tried using the "reading list" feature in the Advanced Options of WebToEpub, but it seems like this feature, although useful in keeping track, still requires the user to re-generate (i.e. repack) every chapter from scratch again in a **new EPUB**. I'm assuming that the original poster of this request was wanting to **update a pre-existing EPUB** with new updated chapters from its source. Perhaps there could be some kind of upload feature (with a storage limit and delete function to avoid clutter if that'd be simpler), where the user **uploads and stores a pre-existing EPUB in the extension itself** with the **file being linked to the source where the chapters were grabbed from** so that it could be updated with **new chapters** rather than having to re-generate a new EPUB and writing in the metadata and redoing the formatting for the chapter links and titles each time the user wants to add in the new chapters for the EPUB.

This is how I understood their request and elaborated a little more on how it could work, but I may be wrong ahah.

**Comment by dteviot** [comment]
@vanessabluellie 
Thanks for responding.
Unfortunately, for security reasons, it's not really possible for an extension to open a file on disk by itself.  A user has to drop the file into the browser.
What could be done is drop an epub into WebToEpub and have it look for additional chapters.
I was thinking of doing this long ago, and WebToEpub packs additional metadata into the epub to allow this to be done.
However, it would take a lot of work to finish implementing.  (Code to unpack/repack an epub would need to be written.)
And, as I've got a "MergeWebToEpub" tool, I am not willing to do the work.
Basically, if you're using the reading list, you can generate a new epub with just the new chapters.  Then use "MergeWebToEpub" to merge the new and old epubs.

**Comment by vanessabluellie** [comment]
@dteviot 

That's completely understandable since this does seem very time-consuming to implement given that there are a couple other tools to assist in doing this to some degree already. Still, thank you for considering it!

**Comment by gamebeaker** [comment]
is implemented with the [library](https://github.com/dteviot/WebToEpub/wiki/Library) function
